docs/src/formula.md

```diff
@@ -14,167 +14,184 @@ fields with possibly heterogeneous types.  One of the primary goals of
+top level) `~`.
+
+!!! note 
+
+    The `@formula` macro **must** be called with parentheses to ensure that
+    the formula is parsed properly.
+
+### An example
+
+```@repl 1
+using Random; Random.seed!(1); # hide
+using StatsModels, DataFrames
+f = @formula(y ~ 1 + a + b + c + b&c)
+df = DataFrame(y = rand(9), a = [1:9;], b = rand(9), c = repeat(["d","e","f"], 3))
+f = apply_schema(f, schema(f, df))
+response, predictors = model_cols(f, df); 
+predictors
+coefnames(f)
 ```
 
-Note that the `@formula` macro **must** be called with parentheses to ensure that
-the formula is parsed properly.
-
-The left side of a formula conventionally represents *dependent* variables, and
-the right side *independent* variables (or regressors).  *Terms* are separated
-by `+`.  Basic terms are the integers `1` or `0`—evaluated as the presence or
-absence of a constant intercept term, respectively—and variables like `x`,
-which will evaluate to the data source column with that name as a symbol (`:x`).
-
-Individual variables can be combined into *interaction terms* with `&`, as in
-`a&b`, which will evaluate to the product of the columns named `:a` and `:b`.
-If either `a` or `b` are categorical, then the interaction term `a&b` generates
-all the product of each pair of the columns of `a` and `b`.
-
-It's often convenient to include main effects and interactions for a number of
-variables.  The `*` operator does this, expanding in the following way:
-
-```jldoctest
-julia> Formula(StatsModels.Terms(@formula(y ~ 1 + a*b)))
-Formula: y ~ 1 + a + b + a & b
+Let's break down the formula expression ` y ~ 1 + a + b + c + b&c`:
+
+At the top level is the **formula separator** `~`, which separates the left-hand
+(or response) variables `y` from the right-hand size (or predictor) variables on
+the right `1 + a + b + c + b&c`.
+
+The left-hand side has one term `y` which means that the resopnse variable is
+the column from the data named `:y`.  The response can be accessed with the
+analogous `response(f, df)` function.
+
+The right hand side is made up of a number of different **terms**, separated by
+`+`: `1 + a + b + c + b&c`.  Each term corresponds to one or more columns in the
+generated model matrix: 
+
+* The first term `1` generates a constant or "intercept" column full of `1.0`s.
+* The next two terms `a` and `b` correspond to columns from the data table
+  called `:a`, `:b`, which both hold numeric data (`Float64` and `Int`
+  respectively).  Such **continuous terms** are converted to `Float64` and
+  copied to the model matrix
+* The term `c` corresponds to the `:c` column in the table, which is _not_
+  numeric, so it has been [contrast coded](@ref Modeling-categorical-data):
+  there are three unique values or levels, and the default coding scheme
+  ([`DummyCoding`](@ref)) generates an indicator variable for each level after
+  the first (e.g., `df[:c] .== "b"` and `df[:c] .== "a"`).
```

Comment:
```suggestion
  the first, e.g. `df[:c] .== "b"` and `df[:c] .== "a"`).
```

Reply:
did you mean to delete that paren there?

Reply:
Yes, there's one opening parenthesis without a closing one in the sentence.